SonarQube.TeamBuild.Integration/CoverageReportDownloader.cs

```diff
@@ -32,17 +40,44 @@ public bool DownloadReport(string reportUrl, string newFullFileName, ILogger log
             string downloadDir = Path.GetDirectoryName(newFullFileName);
             Utilities.EnsureDirectoryExists(downloadDir, logger);
 
-            using (WebClient myWebClient = new WebClient())
-            {
-                myWebClient.UseDefaultCredentials = true;
+            InternalDownloadReport(tfsUri, reportUrl, newFullFileName, logger);
+
+            return true;
+
+        }
 
-                logger.LogMessage(Resources.DOWN_DIAG_DownloadCoverageReportFromTo, reportUrl, newFullFileName);
+        private void InternalDownloadReport(string tfsUri, string reportUrl, string reportDestinationPath, ILogger logger)
+        {
+            VssCredentials vssCreds = null;
+            Uri tfsCollectionUri = new Uri(tfsUri);
+
+            using (TfsTeamProjectCollection collection = TfsTeamProjectCollectionFactory.GetTeamProjectCollection(tfsCollectionUri))
```

Comment:
I've tested with TFS 2013 OM and VSO. I am continuing testing with TFS 2013 OM + TFS 2015 on prem and TFS 2015 OM + VSO (which I expect it will fail)